SetHandleInformation wrong 'int' to 'HANDLE' casts with VC++ using cpp mode.

### Repro case
```
nim cpp --cc:vcc .\src\main.nim
```
```nim
import net

let socket = newSocket(
  Domain.AF_INET,
  SockType.SOCK_DGRAM,
  Protocol.IPPROTO_UDP,
  buffered = false
)
```

### Current Output
```
C:\Users\me\.choosenim\toolchains\nim-#devel\lib\system\io.nim(342): error C2664: 'BOOL SetHandleInformation(HANDLE,DWORD,DWORD)': cannot convert argument 1 from 'int' to 'HANDLE'
C:\Users\me\.choosenim\toolchains\nim-#devel\lib\system\io.nim(342): note: Conversion from integral type to pointer type requires reinterpret_cast, C-style cast or function-style cast
C:\Program Files (x86)\Windows Kits\8.1\include\um\handleapi.h(88): note: see declaration of 'SetHandleInformation'
```

### Expected Output
It compiles.

### Possible Solution

If you just do `c` it works fine:
```
nim c --cc:vcc .\src\main.nim
```
```
Hint: 64214 lines; 9.882s; 75.609MiB peakmem; Debug build; ... [SuccessX]
```

Maybe a special case for cpp vs c when using --c:vcc is needed? Casting rules between C and C++ are more strict?

These PRs touched this area recently:
* https://github.com/nim-lang/Nim/pull/14550 @alaviss 
* https://github.com/nim-lang/Nim/pull/14548 @timotheecour
* https://github.com/nim-lang/Nim/pull/14551 @alaviss 

### Additional Information

```
nim -v
Nim Compiler Version 1.3.5 [Windows: amd64]
Compiled at 2020-07-14
Copyright (c) 2006-2020 by Andreas Rumpf

active boot switches: -d:release
```

--- a/lib/system/io.nim
+++ b/lib/system/io.nim
@@ -283,10 +283,10 @@ elif defined(windows):
     importc: "_get_osfhandle", header: "<io.h>".}
 
   type
-    Handle = distinct pointer
+    IoHandle = distinct pointer
       ## Windows' HANDLE type. Defined as an untyped pointer but is **not**
-      ## one.
-  proc setHandleInformation(handle: Handle, mask, flags: culong): cint {.
+      ## one. Named like this to avoid collision with other `system` modules.
+  proc setHandleInformation(handle: IoHandle, mask, flags: culong): cint {.
     importc: "SetHandleInformation", header: "<handleapi.h>".}
 
 const
@@ -343,7 +343,7 @@ when defined(nimdoc) or (defined(posix) and not defined(nimscript)) or defined(w
       flags = if inheritable: flags and not FD_CLOEXEC else: flags or FD_CLOEXEC
       result = c_fcntl(f, F_SETFD, flags) != -1
     else:
-      result = setHandleInformation(cast[Handle](f), HANDLE_FLAG_INHERIT,
+      result = setHandleInformation(cast[IoHandle](f), HANDLE_FLAG_INHERIT,
                                     culong inheritable) != 0
 
 proc readLine*(f: File, line: var TaintedString): bool {.tags: [ReadIOEffect],

--- a/lib/system/io.nim
+++ b/lib/system/io.nim
@@ -282,7 +282,11 @@ elif defined(windows):
   proc getOsfhandle(fd: cint): int {.
     importc: "_get_osfhandle", header: "<io.h>".}
 
-  proc setHandleInformation(handle: int, mask, flags: culong): cint {.
+  type
+    Handle = distinct pointer
+      ## Windows' HANDLE type. Defined as an untyped pointer but is **not**
+      ## one.
+  proc setHandleInformation(handle: Handle, mask, flags: culong): cint {.
     importc: "SetHandleInformation", header: "<handleapi.h>".}
 
 const
@@ -339,7 +343,8 @@ when defined(nimdoc) or (defined(posix) and not defined(nimscript)) or defined(w
       flags = if inheritable: flags and not FD_CLOEXEC else: flags or FD_CLOEXEC
       result = c_fcntl(f, F_SETFD, flags) != -1
     else:
-      result = setHandleInformation(f.int, HANDLE_FLAG_INHERIT, culong inheritable) != 0
+      result = setHandleInformation(cast[Handle](f), HANDLE_FLAG_INHERIT,
+                                    culong inheritable) != 0
 
 proc readLine*(f: File, line: var TaintedString): bool {.tags: [ReadIOEffect],
               benign.} =

